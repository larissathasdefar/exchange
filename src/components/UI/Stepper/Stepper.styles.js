import styled from 'styled-components'
import MobileStepper from '@material-ui/core/MobileStepper'

export const Container = styled.div`
  min-height: 150px;
  display: flex;
  align-items: stretch;
  justify-content: center;
  flex-direction: column;
`

export const Content = styled.div`
  padding: 12px 0px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`
export const StepContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

export const Progress = styled(MobileStepper)`
  background-color: transparent !important;
  justify-content: center !important;
`
