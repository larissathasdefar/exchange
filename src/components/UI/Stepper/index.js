import React, { PureComponent, Fragment } from 'react'
import _ from 'prop-types'
import Button from '@material-ui/core/Button'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft'
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight'
import Typography from '@material-ui/core/Typography'
import {
  Container,
  Content,
  StepContainer,
  Progress,
} from './Stepper.styles'

class CustomStepper extends PureComponent {
  state = {
    activeStep: this.props.start,
  }

  componentDidUpdate() {
    const { steps, onChangeStep } = this.props
    const { activeStep } = this.state
    return onChangeStep
      ? onChangeStep(activeStep, steps[activeStep])
      : null
  }

  handleNext = () => {
    const { activeStep } = this.state
    const { steps } = this.props
    const nextStep = activeStep === steps.length - 1
      ? 0
      : activeStep + 1
    this.setState({ activeStep: nextStep })
  }

  handleBack = () => {
    const { activeStep } = this.state
    const { steps } = this.props
    const nextStep = activeStep === 0
      ? steps.length - 1
      : activeStep - 1
    this.setState({ activeStep: nextStep })
  }

  renderActualStep = () => {
    const { steps, align } = this.props
    const { activeStep } = this.state
    return (
      <Content>
        <Typography align={align} variant="h4">
          { steps[activeStep].title }
        </Typography>
        <Typography align={align} variant="caption">
          { steps[activeStep].subtitle }
        </Typography>
      </Content>
    )
  }

  render() {
    const { activeStep } = this.state
    const { steps, renderStep } = this.props
    const maxSteps = steps.length
    return (
      <Container>
        <StepContainer>
          <Button onClick={this.handleBack}>
            <KeyboardArrowLeft />
          </Button>
          {
            renderStep
              ? renderStep(steps[activeStep])
              : this.renderActualStep()
          }
          <Button onClick={this.handleNext}>
            <KeyboardArrowRight />
          </Button>
        </StepContainer>
        <Progress
          steps={maxSteps}
          position="static"
          activeStep={activeStep}
        />
      </Container>
    )
  }
}

CustomStepper.defaultProps = {
  align: 'inherit',
  start: 0,
  renderStep: undefined,
  onChangeStep: undefined,
}

CustomStepper.propTypes = {
  steps: _.arrayOf(
    _.shape({
      title: _.string.isRequired,
      subtitle: _.string.isRequired,
    }),
  ).isRequired,
  align: _.string,
  renderStep: _.func,
  start: _.number,
  onChangeStep: _.func,
}

export default CustomStepper
