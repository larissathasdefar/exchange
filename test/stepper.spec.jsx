import React from 'react'
import { shallow } from 'enzyme'

import Stepper from 'ui/Stepper'

describe('Stepper', () => {
  it('Basic example', () => {
    const component = shallow(
      <Stepper
        steps={[
          { title: 'First', subtitle: 'Slide' },
          { title: 'Second', subtitle: 'Slide' },
          { title: 'Third', subtitle: 'Slide' }
        ]}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })

  it('Starting at Second item', () => {
    const component = shallow(
      <Stepper
        steps={[
          { title: 'First', subtitle: 'Slide' },
          { title: 'Second', subtitle: 'Slide' },
          { title: 'Third', subtitle: 'Slide' }
        ]}
        start={1}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })
})