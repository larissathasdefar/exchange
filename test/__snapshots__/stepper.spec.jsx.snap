// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Basic example 1`] = `
"<styled.div>
  <styled.div>
    <Styled(WithStyles(Button)) onClick={[Function]}>
      <pure(KeyboardArrowLeftIcon) style={[undefined]} />
    </Styled(WithStyles(Button))>
    <styled.div>
      <WithStyles(Typography) align=\\"inherit\\" variant=\\"h4\\" style={[undefined]}>
        First
      </WithStyles(Typography)>
      <WithStyles(Typography) align=\\"inherit\\" variant=\\"caption\\" style={[undefined]}>
        Slide
      </WithStyles(Typography)>
    </styled.div>
    <Styled(WithStyles(Button)) onClick={[Function]}>
      <pure(KeyboardArrowRightIcon) style={[undefined]} />
    </Styled(WithStyles(Button))>
  </styled.div>
  <Styled(WithStyles(MobileStepper)) steps={3} position=\\"static\\" activeStep={0} />
</styled.div>"
`;

exports[`Stepper Starting at Second item 1`] = `
"<styled.div>
  <styled.div>
    <Styled(WithStyles(Button)) onClick={[Function]}>
      <pure(KeyboardArrowLeftIcon) style={[undefined]} />
    </Styled(WithStyles(Button))>
    <styled.div>
      <WithStyles(Typography) align=\\"inherit\\" variant=\\"h4\\" style={[undefined]}>
        Second
      </WithStyles(Typography)>
      <WithStyles(Typography) align=\\"inherit\\" variant=\\"caption\\" style={[undefined]}>
        Slide
      </WithStyles(Typography)>
    </styled.div>
    <Styled(WithStyles(Button)) onClick={[Function]}>
      <pure(KeyboardArrowRightIcon) style={[undefined]} />
    </Styled(WithStyles(Button))>
  </styled.div>
  <Styled(WithStyles(MobileStepper)) steps={3} position=\\"static\\" activeStep={1} />
</styled.div>"
`;
